lua/tardis/screens/cl_screen_music.lua

```diff
@@ -1,43 +1,265 @@
+	play_stop_button:SetPos(midX, gap + listT - bT * 1.3)
+	play_stop_button:SetText("Play / Stop")
+	play_stop_button:SetFont(TARDIS:GetScreenFont(screen, "Default"))
+
+	local save_custom_button=vgui.Create("DButton",frame)
+	save_custom_button:SetSize(bW, bT)
+	save_custom_button:SetPos(midX, 3 * gap + 2 * tbT)
+	save_custom_button:SetText("Save")
+	save_custom_button:SetFont(TARDIS:GetScreenFont(screen, "Default"))
+
+	local remove_custom_button=vgui.Create("DButton",frame)
+	remove_custom_button:SetSize(bW, bT)
+	remove_custom_button:SetPos(midX + gap + bW, 3 * gap + 2 * tbT)
+	remove_custom_button:SetText("Remove")
+	remove_custom_button:SetFont(TARDIS:GetScreenFont(screen, "Default"))
+
+--------------------------------------------------------------------------------
+-- Loading data
+--------------------------------------------------------------------------------
+	local url = ""
+
+	for k,v in pairs(default_music) do
+		list_premade:AddLine(v[1])
+	end
+
+	function list_custom:UpdateAll()
+		self:Clear()
+		if url_bar ~= nil then
+			for k,v in pairs(custom_music) do
+				self:AddLine(v[1])
+			end
+		end
+	end
+
+	list_custom:UpdateAll()
+
+--------------------------------------------------------------------------------
+-- Selecting the rows
+--------------------------------------------------------------------------------
+
+	function list_custom:OnRowSelected(rowIndex,row)
+		list_premade:ClearSelection()
+		url_bar:SetText(custom_music[rowIndex][2])
+		name_bar:SetText(custom_music[rowIndex][1])
+		url_bar:SetTextColor(Color(0,0,0))
+		name_bar:SetTextColor(Color(0,0,0))
+	end
+
+	function list_premade:OnRowSelected(rowIndex, row)
+		list_custom:ClearSelection()
+		url = ("https://mattjeanes.com/data/tardis/" .. default_music[rowIndex][2] ..".mp3")
+		url_bar:SetTextColor(Color(139,139,139))
+		name_bar:SetTextColor(Color(139,139,139))
+	end
+
+	local function highlight_custom()
+		url_bar:SetTextColor(Color(0,0,0))
+		name_bar:SetTextColor(Color(0,0,0))
+		list_premade:ClearSelection()
+		list_custom:ClearSelection()
+	end
+
+	function url_bar:OnGetFocus()
+		highlight_custom()
+	end
+	function url_bar:OnValueChange()
+		highlight_custom()
+	end
+	function name_bar:OnGetFocus()
+		highlight_custom()
+	end
+	function name_bar:OnValueChange()
+		highlight_custom()
+	end
+
+--------------------------------------------------------------------------------
+-- Add / remove custom music
+--------------------------------------------------------------------------------
+
+	function name_bar:OnEnter()
+		TARDIS:AddCustomMusic(name_bar:GetText(), url_bar:GetText())
+		list_custom:UpdateAll()
+	end
+
+	function save_custom_button:DoClick()
+		TARDIS:AddCustomMusic(name_bar:GetText(), url_bar:GetText())
+		list_custom:UpdateAll()
+	end
+
+	function remove_custom_button:DoClick()
+		local line = list_custom:GetSelectedLine()
+		if not line then
+			TARDIS:ErrorMessage(LocalPlayer(), "Nothing has been chosen for removal.")
```

Comment:
Remove full stop at end

```suggestion
			TARDIS:ErrorMessage(LocalPlayer(), "Nothing has been chosen for removal")
```